va/va.go

```diff
@@ -471,79 +472,77 @@ func (va *ValidationAuthorityImpl) performRemoteValidation(
 		}).Observe(va.clk.Since(start).Seconds())
 	}()
 
-	type rvaResult struct {
-		hostname string
-		response *vapb.ValidationResult
-		err      error
+	type response struct {
+		addr   string
+		result *vapb.ValidationResult
+		err    error
 	}
 
-	results := make(chan *rvaResult, len(va.remoteVAs))
-
-	for _, i := range rand.Perm(len(va.remoteVAs)) {
-		remoteVA := va.remoteVAs[i]
-		go func(rva RemoteVA, out chan<- *rvaResult) {
+	responses := make(chan *response, remoteVACount)
+	for _, i := range rand.Perm(remoteVACount) {
+		go func(rva RemoteVA, out chan<- *response) {
 			res, err := rva.PerformValidation(ctx, req)
-			out <- &rvaResult{
-				hostname: rva.Address,
-				response: res,
-				err:      err,
+			out <- &response{
+				addr:   rva.Address,
+				result: res,
+				err:    err,
 			}
-		}(remoteVA, results)
+		}(va.remoteVAs[i], responses)
 	}
 
-	required := len(va.remoteVAs) - va.maxRemoteFailures
-	good := 0
-	bad := 0
+	required := remoteVACount - va.maxRemoteFailures
+	var passed []string
+	var failed []string
 	var firstProb *probs.ProblemDetails
 
-	for res := range results {
+	for resp := range responses {
 		var currProb *probs.ProblemDetails
 
-		if res.err != nil {
-			bad++
+		if resp.err != nil {
+			failed = append(failed, resp.result.Perspective)
```

Comment:
This line is failing with a nil pointer dereference in CI. I'm guessing `resp.result` is `nil` when there's an error. Did you run into that issue in #7794?

Reply:
Oh, yeah I did, and we use `resp.Addr` in that case.

Reply:
Updated, and I also ported over similar comments to make this easier to match up.

Reply:
Thanks for the fix. I think now we need a comment on `failed` indicating that it contains a list of strings that are either a perspective or a VA address. Other than that, I think this is fine for now, though I'd rather log the perspective in this error case - that's part of the motivation behind having the primary know what its backend perspectives are in #7817. IMO, we should add the comment, land this, then proceed with followups.